src/khoj/interface/web/chat.html

```diff
@@ -563,6 +564,148 @@
         }
 
+        function openFileBrowser() {
+            event.preventDefault();
+            var overlayText = document.getElementById("dropzone-overlay");
+            var dropzone = document.getElementById('chat-body');
+
+            if (overlayText == null) {
+                dropzone.classList.add('dragover');
+                var overlayText = document.createElement("div");
+                overlayText.innerHTML = "Select a file to share it with Khoj";
+                overlayText.className = "dropzone-overlay";
+                overlayText.id = "dropzone-overlay";
+                dropzone.appendChild(overlayText);
+            }
+
+            const fileInput = document.createElement('input');
+            fileInput.type = 'file';
+            fileInput.addEventListener('change', function() {
+                const selectedFile = fileInput.files[0];
+                uploadDataForIndexing(selectedFile);
+            });
+
+            // Remove overlay text after file input is closed
+            fileInput.addEventListener('blur', function() {
+                dropzone.classList.remove('dragover');
+                var overlayText = document.getElementById("dropzone-overlay");
+                if (overlayText != null) {
+                    overlayText.remove();
+                }
+            });
+
+            // Remove overlay text if file input is cancelled
+            fileInput.addEventListener('cancel', function() {
+                dropzone.classList.remove('dragover');
+                var overlayText = document.getElementById("dropzone-overlay");
+                if (overlayText != null) {
+                    overlayText.remove();
+                }
+            });
+
+            fileInput.click();
+        }
+
+        function uploadDataForIndexing(file) {
+            if (!allowedExtensions.includes(file.type)) {
+                alert("Sorry, that file type is not yet supported");
+                var overlayText = document.getElementById("dropzone-overlay");
+                if (overlayText != null) {
+                    overlayText.remove();
+                }
+                return;
+            }
+
+            const fileName = file.name;
+            var fileContents = null;
+
+            var reader = new FileReader();
+            const formData = new FormData();
+
+            var dropzone = document.getElementById('chat-body');
+
+            var overlayText = document.getElementById("dropzone-overlay");
+            if (overlayText != null) {
+                // Display loading spinner
+                var loadingSpinner = document.createElement("div");
+                overlayText.innerHTML = "Uploading file for indexing";
+                loadingSpinner.className = "spinner";
+                overlayText.appendChild(loadingSpinner);
+            }
+
+            reader.onload = function (event) {
+                fileContents = event.target.result;
+                console.log(fileContents);
```

Comment:
nit: can we remove/dedupe all these `console.log` statements? I'm guessing it was only used to debug file uploads for indexing